cmd/kubeadm/app/phases/patchnode/patchnode_test.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package patchnode
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+)
+
+func TestAnnotateCRISocket(t *testing.T) {
+	tests := []struct {
+		name                       string
+		currentCRISocketAnnotation string
+		newCRISocketAnnotation     string
+		expectedPatch              string
+	}{
+		{
+			"CRI-socket annotation missing",
+			"",
+			"/run/containerd/containerd.sock",
+			"{\"metadata\":{\"annotations\":{\"kubeadm.alpha.kubernetes.io/cri-socket\":\"/run/containerd/containerd.sock\"}}}",
+		},
+		{
+			"CRI-socket annotation already exists",
+			"/run/containerd/containerd.sock",
+			"/run/containerd/containerd.sock",
+			"{}",
+		},
+		{
+			"CRI-socket annotation needs to be updated",
+			"/var/run/dockershim.sock",
+			"/run/containerd/containerd.sock",
+			"{\"metadata\":{\"annotations\":{\"kubeadm.alpha.kubernetes.io/cri-socket\":\"/run/containerd/containerd.sock\"}}}",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+
+			hostname, err := kubeadmutil.GetHostname("")
```

Comment:
it is better to use a dummy hostname instead of querying the OS here - e.g. `foo`